tests/test_sso.py

```diff
@@ -417,190 +268,231 @@ def test_get_profile_and_token_returns_expected_workosprofile_object(
-    ):
-        mock_request_method("get", mock_connections, 200)
+        connections = self.sso.list_connections()
 
-        with pytest.raises(
-            ValueError, match="'connection_type' must be a member of ConnectionType"
-        ):
-            self.sso.list_connections(connection_type="UnknownSAML")
+        assert list_data_to_dicts(connections.data) == mock_connections["data"]
 
-    def test_list_connections_with_connection_type_as_string(
-        self, setup_with_client_id, mock_connections, capture_and_mock_request
+    def test_list_connections_with_connection_type(
+        self, mock_connections, capture_and_mock_http_client_request
     ):
-        request_args, request_kwargs = capture_and_mock_request(
-            "get", mock_connections, 200
+        request_args, request_kwargs = capture_and_mock_http_client_request(
+            http_client=self.http_client,
+            response_dict=mock_connections,
+            status_code=200,
         )
 
-        connections_response = self.sso.list_connections(connection_type="GenericSAML")
+        self.sso.list_connections(connection_type="GenericSAML")
 
         request_params = request_kwargs["params"]
         assert request_params["connection_type"] == "GenericSAML"
 
-    def test_list_connections_with_connection_type_as_enum(
-        self, setup_with_client_id, mock_connections, capture_and_mock_request
-    ):
-        request_args, request_kwargs = capture_and_mock_request(
-            "get", mock_connections, 200
+    def test_delete_connection(self, mock_http_client_with_response):
+        mock_http_client_with_response(
+            self.http_client,
+            status_code=204,
+            headers={"content-type": "text/plain; charset=utf-8"},
         )
 
-        connections_response = self.sso.list_connections(
-            connection_type=ConnectionType.OktaSAML
+        response = self.sso.delete_connection(connection_id="connection_id")
+
+        assert response is None
+
+    def test_list_connections_auto_pagination(
+        self,
+        mock_connections_multiple_data_pages,
+        mock_pagination_request_for_http_client,
+    ):
+        mock_pagination_request_for_http_client(
+            http_client=self.http_client,
+            data_list=mock_connections_multiple_data_pages,
+            status_code=200,
         )
 
-        request_params = request_kwargs["params"]
-        assert request_params["connection_type"] == "OktaSAML"
+        connections = self.sso.list_connections()
+        all_connections = []
 
-    def test_delete_connection(self, setup_with_client_id, mock_raw_request_method):
-        mock_raw_request_method(
-            "delete",
-            "No Content",
-            204,
-            headers={"content-type": "text/plain; charset=utf-8"},
+        for connection in connections.auto_paging_iter():
+            all_connections.append(connection)
```

Comment:
We repeat something like this for every auto paging test. I wonder if it's worth having some helper. (Not in this PR)